doc/releases/changelog-0.10.0.md

```diff
@@ -42,244 +46,229 @@
+  had inadvertently been made more verbose by printing timing information for each run of each pass.
+  This change has been reverted. Instead, the :func:`~.qjit` option `verbose=True` will now instruct
+  the instrumentation to produce this more detailed output.
   [(#1343)](https://github.com/PennyLaneAI/catalyst/pull/1343)
 
-* Stable ABI wheels for Python 3.12 and up are now generated.
-  [(#1357)](https://github.com/PennyLaneAI/catalyst/pull/1357)
-  [(#1385)](https://github.com/PennyLaneAI/catalyst/pull/1385)
-
-* Two new circuit optimization passes, `--disentangle-CNOT` and `--disentangle-SWAP`, are available.
+* Two additional circuit optimizations have been added to Catalyst: `disentangle-CNOT` and
+  `disentangle-SWAP`. The optimizations are available via the :mod:`~.passes` module.
   [(#1154)](https://github.com/PennyLaneAI/catalyst/pull/1154)
 
-  The CNOT pass disentangles CNOT gates whenever possible, e.g., when the control bit is known to be 
-  in the `|0>` state, the pass removes the CNOT. The pass uses a finite state machine to propagate 
-  simple one-qubit states, in order to determine the input states to the CNOT.
-  
-  Similarly, the SWAP pass disentangles SWAP gates whenever possible by using a finite state machine 
-  to propagate simple one-qubit states, similar to the `--disentangle-CNOT` pass.
-
-  Both passes are implemented in accordance with the algorithm from 
-  J. Liu, L. Bello, and H. Zhou, _Relaxed Peephole Optimization: A Novel Compiler Optimization for Quantum Circuits_, 2020, [arXiv:2012.07711](https://arxiv.org/abs/2012.07711) [quant-ph].
-
-* Allow specifying a branch to switch to when setting up a dev environment from the wheels.
-  [(#1406)](https://github.com/PennyLaneAI/catalyst/pull/1406)
+  The optimizations use a finite state machine to propagate limited qubit state information through
+  the circuit to turn CNOT and SWAP gates into cheaper instructions. The pass is based on the work
+  by J. Liu, L. Bello, and H. Zhou, _Relaxed Peephole Optimization: A Novel Compiler Optimization
+  for Quantum Circuits_, 2020, [arXiv:2012.07711](https://arxiv.org/abs/2012.07711).
 
 <h3>Breaking changes 💔</h3>
 
-* The `sample` and `counts` measurement primitives now support dynamic shot values across Catalyst, 
-  although, on the PennyLane side, the device's shots is still constrained to a static integer 
-  literal.
-  [(#1310)](https://github.com/PennyLaneAI/catalyst/pull/1310)
+* The minimum supported PennyLane version has been updated `v0.40`, backwards compatibility in
+  either direction is not maintained.
+  [(#1308)](https://github.com/PennyLaneAI/catalyst/pull/1308)
 
-  To support this, `SampleOp` and `CountsOp` in MLIR no longer carry the shots attribute, since 
-  integer attributes are tied to literal values and must be static.
+* (Device Developers Only) The way the `shots` parameter is initialized in C++ device backends is
```

Comment:
These two entries are not breaking changes for users, but they are breaking changes in the public device api, so I indicated this with `(Device Developers Only)`.